packages/flutter/lib/src/material/scrollbar_theme.dart

```diff
@@ -306,3 +331,57 @@ class ScrollbarTheme extends InheritedWidget {
   @override
   bool updateShouldNotify(ScrollbarTheme oldWidget) => data != oldWidget.data;
 }
+
+/// A scrollbar button style that use [ScrollbarButtonStateColors].
+///
+/// The default value follows the Windows platform style.
+class ScrollbarButtonStateStyles extends ScrollbarButtonStyles {
+  /// Creates a scrollbar button styles that can be used for
+  /// [ScrollbarThemeData.buttonStyles] or [Scrollbar.buttonStyles].
+  const ScrollbarButtonStateStyles({
+    ScrollbarButtonLocation location = ScrollbarButtonLocation.split,
+    double extent = 18.0,
+    double indicatorWidth = 8.0,
+    PaintingStyle indicatorPaintingStyle = PaintingStyle.stroke,
+    ScrollbarButtonStateColors leadingButtonColors = const ScrollbarButtonStateColors(),
+    ScrollbarButtonStateColors trailingButtonColors = const ScrollbarButtonStateColors(),
+  }) : super(
+    location: location,
+    extent: extent,
+    indicatorWidth: indicatorWidth,
+    indicatorPaintingStyle: indicatorPaintingStyle,
+    leadingButtonColors: leadingButtonColors,
+    trailingButtonColors: trailingButtonColors,
+  );
+}
+
+/// A scrollbar button colors style that supports customized colors with
+/// different states.
+///
+/// Only the [Scrollbar] widget responds to this.
+///
+/// The default value follows the Windows platform style.
+class ScrollbarButtonStateColors extends ScrollbarButtonColors {
+  /// Creates a scrollbar button color styles that can be used for
+  /// [ScrollbarButtonStateStyles].
+  const ScrollbarButtonStateColors({
+    Color? backgroundColor,
+    this.hoveredBackgroundColor = const Color(0xFFD2D2D2),
+    this.pressedBackgroundColor = const Color(0xFF787878),
+    Color indicatorColor = const Color(0xFF505050),
+    this.inactiveIndicatorColor = const Color(0xFFA3A3A3),
+  }) : super(
+    backgroundColor: backgroundColor,
+    indicatorColor: indicatorColor,
+  );
+
+  /// The color of the button background when hovered.
+  final Color? hoveredBackgroundColor;
+
+  /// The color of the button background when pressed.
+  final Color? pressedBackgroundColor;
+
+  /// The color of the button indicator when the thumb reaches the
+  /// corresponding edge.
+  final Color? inactiveIndicatorColor;
```

Comment:
These should be wrapped up in a MaterialStateProperty rather than having every state combination enumerated.

Reply:
Perhaps ScrollbarButtonStyle itself should be the return value as a MaterialStateProperty.
It just having a background color and indicator color, and it returns the current colors based on the state.
This is complicated stuff, I really appreciate you continuing to work at this. I think this widget is particularly difficult because it is adaptive, yet the state properties only exist in the material library. I've reached out to @HansMuller to consult on the right way to approach the theming here with state properties.